lib/fog/azurerm/models/storage/file.rb

```diff
@@ -0,0 +1,149 @@
+module Fog
+  module Storage
+    class AzureRM
+      # This class is giving implementation of create/save and
+      # delete/destroy for Blob.
+      class File < Fog::Model
+        identity  :key, aliases: %w(Name name)
+        attribute :accept_ranges
+        attribute :cache_control
+        attribute :committed_block_count
+        attribute :content_length
+        attribute :content_type
+        attribute :content_md5
+        attribute :content_encoding
+        attribute :content_language
+        attribute :content_disposition
+        attribute :copy_completion_time
+        attribute :copy_status
+        attribute :copy_status_description
+        attribute :copy_id
+        attribute :copy_progress
+        attribute :copy_source
+        attribute :directory
+        attribute :etag
+        attribute :lease_duration
+        attribute :lease_state
+        attribute :lease_status
+        attribute :last_modified
+        attribute :metadata
+        attribute :sequence_number
+        attribute :blob_type
+
+        def save(options = {})
+          requires :key
+          requires :directory
+          merge_attributes(File.parse(service.upload_block_blob_from_file(directory, key, options[:file_path], options)))
```

Comment:
storage_service.blobs.create() method is not working because 'file_path' is always passed as nil. The reason is that when we pass a hash in .create method then fog automatically maps this hash to model attributes and finally calls "save" method of that model.